### PR TITLE
Fix unrealistic test variables in KafkaSupervisorTest and tidy up unused variable in checkpointing process

### PR DESCRIPTION
**Pull request by jihoonson**
The start sequence numbers would never have exclusivePartitions in Kafka. Having exclusivePartitions in unit tests wouldn't harm, but would never happen in production. Also, the second parameter for `checkpoint()` is not being used. I removed it.

Since this is not a bug fix, but just tidying up unused variables and test variables, it doesn't have to be backported.

### CODE REVIEW

**Comment by fjy** [comment]
@jihoonson mind fixing conflicts?

**Comment by jihoonson** [comment]
@fjy sure fixed.

**Comment by jihoonson** [comment]
Will fix this.

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java [hunk 1]
```diff
@@ -338,19 +339,16 @@ public void handle()
...
     private final SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType> previousCheckpoint;
-    private final SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType> currentCheckpoint;
 
-    public CheckpointNotice(
+    CheckpointNotice(
         @Nullable Integer nullableTaskGroupId,
         @Deprecated String baseSequenceName,
-        SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType> previousCheckpoint,
-        SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType> currentCheckpoint
+        SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType> previousCheckpoint
```
[the remark this answers is not in the document]
Comment (jihoonson) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java]: Thanks, fixed.

**Comment by jihoonson** [comment]
@fjy @clintropolis @surekhasaharan thanks for the review!

**Pull request merged**